board.go

```diff
@@ -0,0 +1,69 @@
+package honeycombio
+
+// Compile-time proof of interface implementation.
+var _ Boards = (*boards)(nil)
+
+// Boards describes all the boards related methods that Honeycomb supports.
+type Boards interface {
+	// List all boards present in this dataset.
+	List() ([]Board, error)
+
+	// Get a board by its ID. Returns nil, ErrNotFound if there is no board
+	// with the given ID.
+	Get(id string) (*Board, error)
+
+	// Create a new board.
+	Create(data Board) (*Board, error)
```

Comment:
I switched things up a little bit here and didn't create a new type for the created board as the only difference is the ID. Sound reasonable to you?

Also does struct in pointer out make sense?

Reply:
Yeah, I think it's okay to use the same type as parameter and return value. I have set up the Triggers interface similarly.
We should mention this in doc though. Honeycomb did return an error when I tried to create a trigger with an ID: `request failed with status code 400: {"error":"id field not allowed during creation"}`.
https://github.com/kvrhdn/go-honeycombio/blob/main/trigger.go#L19

Returning a pointer makes sense. I'm not sure what the main trade-offs would be, but we would copying  a bit less I think and it is easier for the implementation to just `return nil, err`.

Reply:
>Returning a pointer makes sense. I'm not sure what the main trade-offs would be, but we would copying a bit less I think and it is easier for the implementation to just return nil, err.

I updated this to be pointer in pointer out in 10e303f8a75cd07442c57553f3024f67494843c2 to match Triggerrs

Reply:
>We should mention this in doc though. Honeycomb did return an error when I tried to create a trigger with an ID: request failed with status code 400: {"error":"id field not allowed during creation"}.

Added docs to the `Board` fields in 010a4a9cd362e8fd97196f69c14ba782cf4beae9 and mentioned that this shouldn't be defined for the created.

Also marked `Style` as optional (it has a default).